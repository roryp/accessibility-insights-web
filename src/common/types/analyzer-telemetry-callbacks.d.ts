// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AxeAnalyzerResult } from '../../injected/analyzers/analyzer';
import {
    RuleAnalyzerScanTelemetryData,
    IssuesAnalyzerScanTelemetryData,
    NeedsReviewAnalyzerScanTelemetryData,
} from '../extension-telemetry-events';

export type IAnalyzerTelemetryCallback = ForRuleAnalyzerScanCallback;

export type ForRuleAnalyzerScanCallback = (
    analyzerResult: AxeAnalyzerResult,
    scanDuration: number,
    elementsScanned: number,
    testName: string,
    requirementName?: string,
) => RuleAnalyzerScanTelemetryData;

export type ForIssuesAnalyzerScanCallback = (
    analyzerResult: AxeAnalyzerResult,
    scanDuration: number,
    elementsScanned: number,
    testName: string,
) => IssuesAnalyzerScanTelemetryData;

export type ForNeedsReviewAnalyzerScanCallback = (
    analyzerResult: AxeAnalyzerResult,
    scanDuration: number,
    elementsScanned: number,
    testName: string,
) => NeedsReviewAnalyzerScanTelemetryData;
