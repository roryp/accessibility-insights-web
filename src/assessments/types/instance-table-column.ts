// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentInstanceRowData } from 'DetailsView/components/assessment-instance-table';

export interface InstanceTableColumn {
    key: string;
    name: string;
    onRender: (item: AssessmentInstanceRowData) => JSX.Element;
}
