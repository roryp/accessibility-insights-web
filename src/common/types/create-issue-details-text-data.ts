// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { GuidanceLink } from '../../scanner/rule-to-links-mappings';

export interface CreateIssueDetailsTextData {
    rule: {
        id: string;
        description: string;
        url: string;
        guidance: GuidanceLink[];
    };
    targetApp: {
        name: string;
        url?: string;
    };
    element: {
        identifier: string;
        conciseName: string;
    };
    howToFixSummary: string;
    snippet?: string;
}
