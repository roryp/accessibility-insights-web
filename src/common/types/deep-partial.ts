// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export type DeepPartial<T> = { [P in keyof T]?: DeepPartial<T[P]> };
