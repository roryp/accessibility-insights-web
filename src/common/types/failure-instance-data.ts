// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export type FailureInstanceData = {
    failureDescription?: string;
    path?: string;
    snippet?: string;
};
