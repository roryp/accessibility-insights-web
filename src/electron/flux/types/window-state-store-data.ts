// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export type ViewRoutes = 'deviceConnectView' | 'resultsView';
export type WindowStates = 'customSize' | 'maximized' | 'fullScreen';

export interface WindowStateStoreData {
    routeId: ViewRoutes;
    currentWindowState: WindowStates;
}
