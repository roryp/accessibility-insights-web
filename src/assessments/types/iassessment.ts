// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InitialDataCreator } from 'background/create-initial-assessment-test-data';
import { RequirementOrdering } from 'common/assessment/requirement';
import { AssessmentVisualizationConfiguration } from 'common/configs/assessment-visualization-configuration';
import { AnyExtension } from 'common/extensibility/extension-point';
import { VisualizationType } from 'common/types/visualization-type';
import { ContentPageComponent } from 'views/content/content-page';
import { Requirement } from './requirement';

interface BaseAssessment {
    key: string;
    visualizationType: VisualizationType;
    title: string;
    gettingStarted: JSX.Element;
    guidance?: ContentPageComponent;
    requirements: Requirement[];
    featureFlag?: { required?: string[] };
    requirementOrder?: RequirementOrdering;
    extensions?: AnyExtension[];
    initialDataCreator?: InitialDataCreator;
}

export interface ManualAssessment extends BaseAssessment {}

export interface AssistedAssessment extends BaseAssessment {
    storeDataKey: string;
    visualizationConfiguration?: Partial<AssessmentVisualizationConfiguration>;
}

export interface Assessment extends BaseAssessment {
    getVisualizationConfiguration: () => AssessmentVisualizationConfiguration;
    requirementOrder: RequirementOrdering;
}
