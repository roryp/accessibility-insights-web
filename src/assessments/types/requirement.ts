// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { UniquelyIdentifiableInstances } from 'background/instance-identifier-generator';
import { ManualTestStatus } from 'common/types/manual-test-status';
import {
    AssessmentNavState,
    GeneratedAssessmentInstance,
    InstanceIdToInstanceDataMap,
} from 'common/types/store-data/assessment-result-data';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import {
    AssessmentInstanceRowData,
    AssessmentInstanceTable,
} from 'DetailsView/components/assessment-instance-table';
import { RequirementLink } from 'DetailsView/components/requirement-link';
import { Analyzer } from 'injected/analyzers/analyzer';
import { AnalyzerProvider } from 'injected/analyzers/analyzer-provider';
import { DecoratedAxeNodeResult } from 'injected/scanner-utils';
import {
    PropertyBags,
    VisualizationInstanceProcessorCallback,
} from 'injected/visualization-instance-processor';
import { Drawer } from 'injected/visualization/drawer';
import { DrawerProvider } from 'injected/visualization/drawer-provider';
import { IColumn } from 'office-ui-fabric-react';
import { DictionaryStringTo } from 'types/common-types';
import { ContentPageComponent, HyperlinkDefinition } from 'views/content/content-page';
import { IGetMessageGenerator } from '../assessment-default-message-generator';
import { InstanceTableColumn } from './instance-table-column';
import { ReportInstanceFields } from './report-instance-field';

export interface Requirement {
    key: string;
    description: JSX.Element;
    order?: number;
    name: string;
    renderStaticContent?: () => JSX.Element;
    howToTest: JSX.Element;
    addFailureInstruction?: string;
    infoAndExamples?: ContentPageComponent;
    isManual: boolean;
    // This is for semi-manual cases where we can't present a list of instances like an assisted
    // test would, but can infer a PASS or FAIL state. If not specified, acts like () => UNKNOWN.
    getInitialManualTestStatus?: (instances: InstanceIdToInstanceDataMap) => ManualTestStatus;
    guidanceLinks: HyperlinkDefinition[];
    columnsConfig?: InstanceTableColumn[];
    getAnalyzer?: (provider: AnalyzerProvider) => Analyzer;
    getVisualHelperToggle?: (props: VisualHelperToggleConfig) => JSX.Element;
    // Any results this returns false for will be omitted from visual helper displays, but still
    // present for the purposes of instance lists or getInitialManualTestStatus. By default, all
    // results support visualization.
    isVisualizationSupportedForResult?: (result: DecoratedAxeNodeResult) => boolean;
    visualizationInstanceProcessor?: VisualizationInstanceProcessorCallback<
        PropertyBags,
        PropertyBags
    >;
    getDrawer?: (provider: DrawerProvider, featureFlagStoreData?: FeatureFlagStoreData) => Drawer;
    getNotificationMessage?: (selectorMap: DictionaryStringTo<any>) => string;
    doNotScanByDefault?: boolean;
    switchToTargetTabOnScan?: boolean;
    generateInstanceIdentifier?: (instance: UniquelyIdentifiableInstances) => string;
    reportInstanceFields?: ReportInstanceFields;
    renderReportDescription?: () => JSX.Element;
    getInstanceStatus?: (result: DecoratedAxeNodeResult) => ManualTestStatus;
    getInstanceStatusColumns?: () => Readonly<IColumn>[];
    getDefaultMessage?: IGetMessageGenerator;
    renderInstanceTableHeader?: (
        table: AssessmentInstanceTable,
        items: AssessmentInstanceRowData[],
    ) => JSX.Element;
    renderRequirementDescription?: (requirementLink: RequirementLink) => JSX.Element;
}

export type VisualHelperToggleConfigDeps = {
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
};

export interface VisualHelperToggleConfig {
    deps: VisualHelperToggleConfigDeps;
    assessmentNavState: AssessmentNavState;
    instancesMap: DictionaryStringTo<GeneratedAssessmentInstance>;
    isStepEnabled: boolean;
    isStepScanned: boolean;
}
