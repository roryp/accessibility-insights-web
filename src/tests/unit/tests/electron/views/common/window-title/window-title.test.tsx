// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { WindowStateStoreData } from 'electron/flux/types/window-state-store-data';
import { WindowTitle, WindowTitleProps } from 'electron/views/common/window-title/window-title';
import { PlatformInfo } from 'electron/window-management/platform-info';
import { shallow } from 'enzyme';
import * as React from 'react';
import { IMock, Mock } from 'typemoq';

describe('WindowTitleTest', () => {
    let platformInfoMock: IMock<PlatformInfo>;
    let windowStateStoreData: WindowStateStoreData;
    let props: WindowTitleProps;

    beforeEach(() => {
        platformInfoMock = Mock.ofType(PlatformInfo);
        windowStateStoreData = { currentWindowState: 'maximized', routeId: 'deviceConnectView' };

        props = {
            pageTitle: 'page title 1',
            deps: { platformInfo: platformInfoMock.object },
            children: <span>logo</span>,
            actionableIcons: [<div key="key1">icon1</div>, <div key="key2">icon2</div>],
            windowStateStoreData,
        };
    });

    it('renders without actionable icons', () => {
        props.children = <span>logo</span>;
        props.actionableIcons = undefined;

        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders nothing if fullscreen', () => {
        windowStateStoreData.currentWindowState = 'fullScreen';

        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders with actionable icons', () => {
        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders without actionable buttons for mac', () => {
        platformInfoMock.setup(p => p.isMac()).returns(() => true);

        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders without children & actionable icon', () => {
        props.children = undefined;
        props.actionableIcons = undefined;

        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders with custom class name', () => {
        props.className = 'custom-class-name';

        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders with custom header text class name', () => {
        props.headerTextClassName = 'custom-class-name';

        const rendered = shallow(<WindowTitle {...props} />);

        expect(rendered.getElement()).toMatchSnapshot();
    });
});
