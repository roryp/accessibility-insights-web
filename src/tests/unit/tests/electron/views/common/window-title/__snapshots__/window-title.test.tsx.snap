// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowTitleTest renders nothing if fullscreen 1`] = `null`;

exports[`WindowTitleTest renders with actionable icons 1`] = `
<header
  className="windowTitle"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Accessibility Insights for Android
       - 
      page title 1
    </title>
  </HelmetWrapper>
  <div
    className="titleContainer"
  >
    <span>
      logo
    </span>
    <span
      className="headerText"
    >
      Accessibility Insights
    </span>
  </div>
  <div
    className="actionableIconsContainer"
  >
    <div>
      icon1
    </div>
    <div>
      icon2
    </div>
  </div>
</header>
`;

exports[`WindowTitleTest renders with custom class name 1`] = `
<header
  className="windowTitle custom-class-name"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Accessibility Insights for Android
       - 
      page title 1
    </title>
  </HelmetWrapper>
  <div
    className="titleContainer"
  >
    <span>
      logo
    </span>
    <span
      className="headerText"
    >
      Accessibility Insights
    </span>
  </div>
  <div
    className="actionableIconsContainer"
  >
    <div>
      icon1
    </div>
    <div>
      icon2
    </div>
  </div>
</header>
`;

exports[`WindowTitleTest renders with custom header text class name 1`] = `
<header
  className="windowTitle"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Accessibility Insights for Android
       - 
      page title 1
    </title>
  </HelmetWrapper>
  <div
    className="titleContainer"
  >
    <span>
      logo
    </span>
    <span
      className="headerText custom-class-name"
    >
      Accessibility Insights
    </span>
  </div>
  <div
    className="actionableIconsContainer"
  >
    <div>
      icon1
    </div>
    <div>
      icon2
    </div>
  </div>
</header>
`;

exports[`WindowTitleTest renders without actionable buttons for mac 1`] = `
<header
  className="windowTitle"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Accessibility Insights for Android
       - 
      page title 1
    </title>
  </HelmetWrapper>
  <div
    className="titleContainer"
  >
    <span>
      logo
    </span>
    <span
      className="headerText"
    >
      Accessibility Insights
    </span>
  </div>
</header>
`;

exports[`WindowTitleTest renders without actionable icons 1`] = `
<header
  className="windowTitle"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Accessibility Insights for Android
       - 
      page title 1
    </title>
  </HelmetWrapper>
  <div
    className="titleContainer"
  >
    <span>
      logo
    </span>
    <span
      className="headerText"
    >
      Accessibility Insights
    </span>
  </div>
</header>
`;

exports[`WindowTitleTest renders without children & actionable icon 1`] = `
<header
  className="windowTitle"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Accessibility Insights for Android
       - 
      page title 1
    </title>
  </HelmetWrapper>
  <div
    className="titleContainer"
  >
    <span
      className="headerText"
    >
      Accessibility Insights
    </span>
  </div>
</header>
`;
